server/kv/db_benchmark_deleterange_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2025 StreamNative, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/streamnative/oxia/common"
+	"github.com/streamnative/oxia/proto"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/utils/pointer"
+)
+
+func BenchmarkGenerate100(b *testing.B) {
```

Comment:
This is the benchmark test. if you want to understand more details, you can check out to commit 8996085ca88f2ae394213df55595acdc16cff86f and check the tests/deleterange_test.go . that is using the actual data.

Reply:
Do you have the before vs after results?

Reply:
```
main branch

goos: darwin
goarch: arm64
pkg: github.com/streamnative/oxia/server/kv
cpu: Apple M1 Pro
BenchmarkDeleteRange
BenchmarkDeleteRange-10    	    8863	  34814860 ns/op
PASS

this branch

goos: darwin
goarch: arm64
pkg: github.com/streamnative/oxia/server/kv
cpu: Apple M1 Pro
BenchmarkDeleteRange
BenchmarkDeleteRange-10    	   10000	    848611 ns/op
PASS


```